lti_consumer/plugin/compat.py

```diff
@@ -96,10 +96,34 @@ def publish_grade(block, user, score, possible, only_if_higher=False, score_dele
     )
 
 
-def has_access(*args, **kwargs):
+def user_has_access(user, block, course_key):
     """
-    Import and run the `has_access` method from courseware module.
+    Checks if a user has access to given xblock.
+
+    ``has_access`` doesn't checks for course enrollment. On the otherhand, ``get_course_with_access``
+    only checks for the course itself. There is no way to check access for specific xblock. This function
+    has been created to perform a combination of check for both enrollment and access for specific xblock.
+
+    Args:
+        user: User Object
+        block: xblock Object to check permission for
+        course_key: A course_key specifying which course run this access is for.
+
+    Returns:
+        bool: True if has access, False otherwise.
     """
     # pylint: disable=import-error,import-outside-toplevel
-    from lms.djangoapps.courseware.access import has_access as course_has_access
-    return course_has_access(*args, **kwargs)
+    from lms.djangoapps.courseware.courses import get_course_by_id, check_course_access
+    from lms.djangoapps.courseware.access import has_access
+
+    # get the course
+    course = get_course_by_id(course_key)
+
+    # check if user is authenticated & enrolled
+    course_access = check_course_access(course, user, 'load', check_if_enrolled=True, check_if_authenticated=True)
+
+    # check if user has access to xblock
+    block_access = has_access(user, 'load', block, course_key)
+
+    # return True if have access for both course and xblock
```

Comment:
Nit: Return True if the user has access to xblock and is enrolled in that specific course.